src/inmanta/resources.py

```diff
@@ -257,7 +256,7 @@ def allow_references[T](v: T) -> T:
 
             case dict() | proxy.DictProxy():
                 return {
-                    key: self.collect_references(value, f"{path}.{dict_path.NormalValue(key).escape()}")
+                    key: self.collect_references(value, f"{path}.'{key.replace("'", "\\'")}'")
```

Comment:
```suggestion
                    key: self.collect_references(value, f"{path}.'{key.replace("'", r"\'")}'")
```

Also, quite the nesting here: ' inside "" inside '' inside "". But if it works I'm fine with it.